packages/style-engine/class-wp-style-engine-css-declarations.php

```diff
@@ -113,6 +113,29 @@ public function get_declarations() {
 		return $this->declarations;
 	}
 
+	/**
+	 * Filters a CSS property + value pair.
+	 *
+	 * @param string $property The CSS property.
+	 * @param string $value    The value to be filtered.
+	 * @param string $spacer   The spacer between the colon and the value. Defaults to an empty string.
+	 *
+	 * @return string The filtered declaration as a single string.
+	 */
+	protected static function filter_declaration( $property, $value, $spacer = '' ) {
+		if ( isset( $property ) && isset( $value ) && '' === trim( $spacer ) ) {
+			// Allow a controlled set of `display` values.
+			if (
+				'display' === $property &&
+				in_array( $value, array( 'block', 'inline', 'inline-block', 'flex', 'inline-flex', 'grid', 'inline-grid' ), true )
```

Comment:
Any reason why we couldn't add a `$attrs[] = 'display';` entry to `gutenberg_safe_style_attrs_6_1()` in lib/compat/wordpress-6.1/blocks.php?

If we're not confident that a core patch for display will be in before 6.1 maybe add another `safe_style_css` filter?

Reply:
Over in https://github.com/WordPress/wordpress-develop/pull/2928#discussion_r913299684 we discussed the issue with `display: none`, in that we still want that value to be excluded, but to allow the remaining `display` values.

Currently in the Layout output of the Theme JSON class we have special handling for `display` here: https://github.com/WordPress/gutenberg/blob/trunk/lib/compat/wordpress-6.1/class-wp-theme-json-6-1.php#L1370

For the purposes of this PR, we could always remove the `display` handling, since we're not yet using the style engine for global styles, and defer the issue to later.

I _think_ the question comes down to — should WordPress in general allow `display` with a set number of values, or is it that the Layout / theme.json might need it for a particular use case, but it shouldn't otherwise be more broadly exposed.

Reply:
All that said, in principle, I really like the idea that the style engine could "just work" and outsource CSS validation / sanitization to core functions and filters 🤔

Reply:
> we discussed the issue with display: none, in that we still want that value to be excluded

I've left a comment over in that discussion. It would be good to validate that claim.

> For the purposes of this PR, we could always remove the display handling, since we're not yet using the style engine for global styles, and defer the issue to later.

I always say "if in doubt, chuck it out" 😄 

If we're not using, is there any benefit to having it right now or for 6.1? Or could this be resolved in the core patch?

> I think the question comes down to — should WordPress in general allow display with a set number of values, or is it that the Layout / theme.json might need it for a particular use case, but it shouldn't otherwise be more broadly exposed.

Yeah for sure. Do you think it'd be helpful to ask a wider audience before committing here?

Reply:
> I always say "if in doubt, chuck it out" 😄

> If we're not using it, is there any benefit to having it right now or for 6.1?

Great point. For the scope of this PR, we only need to resolve the issue reported in https://github.com/WordPress/gutenberg/issues/42948, as `display` is still handled in the Theme JSON class. Let's avoid dealing with `display` in this PR until we need to refactor the centralised Layout output.